### ISSUE TITLE
Add a new parameter to iterate over different ACLs for WAF v2

### ISSUE THREAD

**Issue by javiersanchz** [wazuh/wazuh#23363]
|Epic|
|---|
| #23361 | 

## Description

The objective of the following issue is to be able to iterate over the different ACLs, searching for the records generated within a specific ACL.

The path generated by WAF v2 is as follows:

`<bucket_name>/<prefix>/AWSLogs/<suffix>/<account_id>/WAFLogs/<region>/<waf_name>/<year>/<month>/<day>/<hh>/<mm>
`
- Within this path, we find the `<waf_name>` which represents the name of the ACL where these records have been generated

## Task

- Create a new parameter in our AWS module that iterates over a specific ACL passed as a parameter.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by fdalmaup** [comment]
# Issue Update

- Added the new `waf_acls` parameter, modifying the C parser and script code.
- The rest of the modifications to iterate over the ACL folders based on the parameter will be carried out once #23397 is merged into the epic branch.

**Comment by fdalmaup** [comment]
# Issue Update

Due to internal changes in the team, the assignee for this issue will be @javiersanchz. He will continue the work carried out in the [enhancement/23363-add-waf-acl-parameter](https://github.com/wazuh/wazuh/tree/enhancement/23363-add-waf-acl-parameter) branch and perform the necessary testing related to it.

**Comment by javiersanchz** [comment]
## Update

- I reviewed the changes made in the issue branch.

- I performed a rebase on the epic branch that had the code added to process WAF V2 logs included in https://github.com/wazuh/wazuh/issues/22572

- I was testing in an environment to retrieve the logs generated in the bucket using the new `waf_acls` parameter for the ACL name, but I am still encountering errors:

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-V2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: +++ Error: AWSBucket.__init__() got an unexpected keyword argument 'waf_acls'
Unknown error: AWSBucket.__init__() got an unexpected keyword argument 'waf_acls'
Traceback (most recent call last):
  File "/var/ossec/wodles/aws/aws-s3.py", line 200, in <module>
    main(sys.argv[1:])
  File "/var/ossec/wodles/aws/aws-s3.py", line 77, in main
    bucket = bucket_type(reparse=options.reparse,
  File "/var/ossec/wodles/aws/buckets_s3/waf.py", line 33, in __init__
    AWSCustomBucket.__init__(self, **kwargs)
  File "/var/ossec/wodles/aws/buckets_s3/aws_bucket.py", line 739, in __init__
    AWSBucket.__init__(self, db_table_name=db_table_name if db_table_name else 'custom', **kwargs)
TypeError: AWSBucket.__init__() got an unexpected keyword argument 'waf_acls'
```

**Comment by javiersanchz** [comment]
## Update

- I was setting up the environment and seeing very little about what to implement in the code, putting it on hold again to prioritize https://github.com/wazuh/wazuh/issues/24904 for version 4.9.0

**Comment by javiersanchz** [comment]
## Update 

Another file was created within the bucket with a different `acl_name`:

![image](https://github.com/user-attachments/assets/6bf1b393-cd37-4894-893d-c563eb4da7d9)

Once created, the new parameter was tested to filter by the `acl_name`, in this case `-wa AWS-WAF-V2`, as shown below: 

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-V2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
DEBUG: +++ Working on xxxxxxx - us-east-1
DEBUG: +++ Marker: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/10/xxxxxx_waflogs_us-east-1_AWS-WAF-V2_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/567970947422_waflogs_us-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```

It seems to iterate correctly over the provided ACL name and display the logs, but when re-executed with the ACL name changed to the other one created, in this case `AWS-WAF-TEST1`:

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxx - us-east-1
DEBUG: +++ Marker: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/567970947422_waflogs_us-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
```

It shows that there are no logs to process in the bucket and the Marker still indicates the previous ACL name: `...us-east-1/AWS-WAF-V2...`

**Comment by javiersanchz** [comment]
## Update

- The problem occurred when retrieving the marker from the database. It was obtaining the last marker, which caused the issue that even though the `ACL_NAME` was being handled correctly when fetching the logs, the last used `ACL_NAME` was being utilized. With the new changes, this is the output:

-------------------------------------------------------------------------------------------------------------------------------------------------------------------------

`-wa AWS-WAF-V2`

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-V2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
DEBUG: +++ Working on 567970947422 - us-east-1
DEBUG: +++ Marker: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2019/10/22
DEBUG: ++ Found new log: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/10/567970947422_waflogs_us-east-1_AWS-WAF-V2_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/567970947422_waflogs_us-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```

`-wa AWS-WAF-TEST1`

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on 567970947422 - us-east-1
DEBUG: +++ Marker: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-TEST1/2019/10/22
DEBUG: ++ Found new log: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/567970947422_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/567970947422_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```

If we run them again:

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-V2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on 567970947422 - us-east-1
DEBUG: +++ Marker: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/10/567970947422_waflogs_us-east-1_AWS-WAF-V2_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/567970947422_waflogs_us-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: 567970947422/us-east-1
DEBUG: +++ DB Maintenance
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -wa AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on 567970947422 - us-east-1
DEBUG: +++ Marker: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-TEST1/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/567970947422_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/567970947422/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/567970947422_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: 567970947422/us-east-1
DEBUG: +++ DB Maintenance
```

**Comment by javiersanchz** [comment]
## Update

- The changes requested in the review were included
- Tests were conducted again to verify functionality
- Regarding the [request](https://github.com/wazuh/wazuh/pull/24988#discussion_r1700570580) to change the acronym for the new parameter, there was a discussion about changing it or leaving it as it is. Currently, in the issue, the parameter is called `-wa`, but it might be more intuitive to call it `-waf` if possible, or simpler to call it just `-w`
  - It was decided that @fdalmaup  would return to discuss this and make a decision (cc: @GGP1  )

**Comment by javiersanchz** [comment]
## Update

- With respect to the latest change proposed in https://github.com/wazuh/wazuh/pull/24988#pullrequestreview-2215766981, all of this will be taken into account and will be carried out as part of the epic in https://github.com/wazuh/wazuh/issues/23364, where both the documentation about WAF and the new parameter added will be modified.

**Comment by javiersanchz** [comment]
## Update

I'm still working on it because one of the requested changes is that if the `-w` parameter is not provided, it should return all the logs from the different `ACLs` found within the `account_id`

The issue is that the current code was obtaining the `ACL name` through the last key of the logs provided by calling the bucket, which was then used to generate the `full_prefix`

This has been modified to search for and generate a list of all the ACL names it finds. However, when displaying the logs, it only returns the logs for the last ACL name it retrieves, as can be seen here:

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
DEBUG: +++ Working on xxxxxxxx - us-east-1
Nombres de ACL encontrados: ['AWS-WAF-V2', 'AWS-WAF-TEST-MULTIPLE', 'AWS-WAF-TEST1']
Name ACL acl_name: AWS-WAF-V2
DEBUG: +++ Marker: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/10/xxxx_waflogs_us-east-1_AWS-WAF-V2_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/xxxxx_waflogs_us-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```
One of the ideas to solve this is through a `yield` generator, trying to modify the code as little as possible to introduce it. However, it still results in an error when executed

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxx - us-east-1
DEBUG: +++ Marker: AWSLogs/xxxxxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/xxxxxxxx-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ Unexpected error: Parameter validation failed:
Invalid type for parameter Prefix, value: <generator object AWSWAFBucket.get_full_prefix at 0x79a30c1d3920>, type: <class 'generator'>, valid types: <class 'str'>
ERROR: Unexpected error querying/working with objects in S3: Parameter validation failed:
Invalid type for parameter Prefix, value: <generator object AWSWAFBucket.get_full_prefix at 0x79a30c1d3920>, type: <class 'generator'>, valid types: <class 'str'>
```

**Comment by javiersanchz** [comment]
## Update 

I've been changing the way of accessing the ACL names because it wasn't being handled properly. I'm still working on it because it seems that it's not functioning correctly 

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
DEBUG: +++ Working on xxxxxxxx - us-east-1
Nombres de ACL encontrados: ['AWS-WAF-V2', 'AWS-WAF-TEST-MULTIPLE', 'AWS-WAF-TEST1']
DEBUG: +++ Marker: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/10/xxxx_waflogs_us-east-1_AWS-WAF-V2_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-V2/2024/05/03/11/15/xxxxx_waflogs_us-east-1_AWS-WAF-V2_20240503T1115Z_7409afbb.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxx_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxx_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```
As you can see, the logs for the ACL `AWS-WAF-TEST-MULTIPLE` are not being processed. On the other hand, I'm not sure if the `Marker` is entirely correct.

I'm still working on it

**Comment by javiersanchz** [comment]
## Update

- I was discussing with @fdalmaup how our current `aws_buckets` code can handle multiple `get_full_prefix` calls if we provide each of the `acl_names` we obtain. Currently, the code handles only one `get_full_prefix` to filter the data we retrieve.

- Another option would be to try handling each one individually within the waf class before sending it to `build_s3_filter_args`.

- For now, it will be put on hold to address this and proceed.

**Comment by javiersanchz** [comment]
## Update

I am still working on how to iterate over all ACLs to display all logs within that bucket

I first tried passing get_full_prefix without any ACL to use it as a reference in `StartAfter`, but when adding `only_logs_after`, it does not retrieve any logs because the ACL is missing between the `region` and the `date` provided by `only_logs_after`

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxx - us-east-1
Full_prefix : AWSLogs/56xxxxx/WAFLogs/us-east-1/
filter_args of build_s3_filter_args: {'Bucket': 'aws-waf-logs-wodle-v2', 'MaxKeys': 1000, 'Prefix': 'AWSLogs/xxxxxxxx/WAFLogs/us-east-1/'}
filter_args of build_s3_filter_args with StartAfter: {'Bucket': 'aws-waf-logs-wodle-v2', 'MaxKeys': 1000, 'Prefix': 'AWSLogs/xxxxxxxx/WAFLogs/us-east-1/', 'StartAfter': 'AWSLogs/xxxxxxx/WAFLogs/us-
DEBUG: +++ Marker: AWSLogs/xxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: +++ No logs to process in bucket: xxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
```

I tried not passing any ACL name initially and adding them one by one later in `build_s3_filter_args`, but it still doesn't process the logs correctly, as it references only the first ACL name and ignores the rest : 

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxx - us-east-1
Prefix of get_full_prefix AWSLogs/xxxxxxx/WAFLogs/us-east-1/None/
filter_args of build_s3_filter_args with only one prefix: {'Bucket': 'aws-waf-logs-wodle-v2', 'MaxKeys': 1000, 'Prefix': 'AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/None/'}
filter_args of build_s3_filter_args with StartAfter: {'Bucket': 'aws-waf-logs-wodle-v2', 'MaxKeys': 1000, 'Prefix': 'AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/None/', 'StartAfter': 'AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxx_waflogs_us-east-1_AWS-WAF-TEST1_xxxxxxxx_7409afbb.log.gz'}
filter_args of build_s3_filter_args with StartAfter 2: {'Bucket': 'aws-waf-logs-wodle-v2', 'MaxKeys': 1000, 'Prefix': 'AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/None/', 'StartAfter': 'AWSLogs/xxxxxxx/WAFLogs/us-east-1/None/2019/10/22'}
DEBUG: +++ Marker: AWSLogs/xxxxxxxx/WAFLogs/us-east-1/None/2019/10/22
DEBUG: +++ No logs to process in bucket: xxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
```

I am now testing modifications to both `build_s3_filter_args` and `iter_files_in_bucket` within `waf.py` to be able to iterate over all the ACLs

**Comment by javiersanchz** [comment]
## Update

Finally, it was decided to handle AWS WAF V2 through the `-w` parameter, passing the ACL names as arguments to the parameter, similar to how it is done with` --regions`, handling one or more.

A condition was added to start with, so that if no parameter is provided, it correctly distinguishes between whether it is V1 (Kinesis) or V2


Kinesis bucket : 

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket wazuh-aws-wodle-waf  --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
The functionality to process WAF logs stored in S3 via Kinesis was deprecated in 5.0. Consider configuring WAF to store its logs directly in an S3 bucket instead. Check https://documentation.wazuh.com/current/amazon/services/supported-services/waf.html for more information.
DEBUG: +++ Marker: 2019/10/22
DEBUG: ++ Found new log: 2019/10/23/10/aws-waf-logs-delivery-stream-1-2019-10-23-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2019/10/23/11/aws-waf-logs-delivery-stream-1-2019-10-23-11-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake
ERROR: the 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake file doesn't have the expected structure.
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-1
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-2
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-3
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-multiple-values-in-ruleGroupList
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-multiple-values-in-ruleGroupList
DEBUG: +++ DB Maintenance
```


And in the case where the bucket is` WAF V2` and the `-w` parameter is not provided:

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2                    
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxx - us-east-1
DEBUG: +++ WARNING: No waf_acls parameter found
```
A `warning` is issued about the missing `waf_acls` parameter

I’m still working on processing the `-w` parameter as a list, displaying the logs for the specified ACLs

**Comment by javiersanchz** [comment]
I was finishing up the implementation to pass a list of ACLs using the `-w` parameter : 

Execution with a single ACL:

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```
If we run it again: 

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
```

Execution with a two ACLs:

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST-MULTIPLE,AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST-MULTIPLE
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/10/5679xxxxxxxxxx70947422_waflogs_us-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/15/xxxxxxxxxx_waflogs_us-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxxx_waflogs_us-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
```
If we do not pass the parameter `-w`:

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default  --regions us-east-1 --type waf --skip_on_error -d 2                                       
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxx - us-east-1
DEBUG: +++ WARNING: No waf_acls parameter found
```

Without the parameter `only_logs_after`: 

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default  --regions us-east-1,us-east-2 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST-MULTIPLE,AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST-MULTIPLE
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/2024/09/06
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/10/xxxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/15/xxxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/2024/09/06
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
```
Kinesis bucket : 

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket wazuh-aws-wodle-waf  --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
The functionality to process WAF logs stored in S3 via Kinesis was deprecated in 5.0. Consider configuring WAF to store its logs directly in an S3 bucket instead. Check https://documentation.wazuh.com/current/amazon/services/supported-services/waf.html for more information.
DEBUG: +++ Marker: 2019/10/22
DEBUG: ++ Found new log: 2019/10/23/10/aws-waf-logs-delivery-stream-1-2019-10-23-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2019/10/23/11/aws-waf-logs-delivery-stream-1-2019-10-23-11-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake
ERROR: the 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake file doesn't have the expected structure.
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-1
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-2
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-3
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-multiple-values-in-ruleGroupList
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-multiple-values-in-ruleGroupList
DEBUG: +++ DB Maintenance
```

**Comment by javiersanchz** [comment]
## Update

I was monitoring the error that occurred with `AWS WAF Kinesis`: 

```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket wazuh-aws-wodle-waf  --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
The functionality to process WAF logs stored in S3 via Kinesis was deprecated in 5.0. Consider configuring WAF to store its logs directly in an S3 bucket instead. Check https://documentation.wazuh.com/current/amazon/services/supported-services/waf.html for more information.
DEBUG: +++ Marker: 2019/10/22
DEBUG: ++ Found new log: 2019/10/23/10/aws-waf-logs-delivery-stream-1-2019-10-23-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2019/10/23/11/aws-waf-logs-delivery-stream-1-2019-10-23-11-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake
ERROR: the 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake file doesn't have the expected structure.
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-1
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-2
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-3
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-multiple-values-in-ruleGroupList
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-multiple-values-in-ruleGroupList
DEBUG: ++ Found new log: 567970947422/2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake
ERROR: the 567970947422/2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake file doesn't have the expected structure.
DEBUG: ++ Found new log: test_prefix/2019/10/22/10/aws-waf-logs-delivery-stream-1-2019-10-22-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718193
ERROR: the test_prefix/2019/10/22/10/aws-waf-logs-delivery-stream-1-2019-10-22-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718193 file doesn't have the expected structure.
DEBUG: +++ DB Maintenance
```

I haven’t submitted it for review yet because I want to do more testing first.

**Comment by javiersanchz** [comment]
## Update

The error with `Kinesis`, as shown in the previous update, has been fixed

On the other hand, a bug has also been fixed, as the `Marker` appeared twice, as you can see here:

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST-MULTIPLE,AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxx - us-east-1 - ACL: AWS-WAF-TEST-MULTIPLE
DEBUG: +++ Marker: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ Marker: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/10/xxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/15/xxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
DEBUG: +++ Working on xxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ Marker: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxx/us-east-1
```


This was due to inheriting and calling the `build_s3_filter_args` function from the parent class, which was returning the first `Marker`: `Marker: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz`, which is malformed.

The second `Marker` is the one we find in `build_s3_filter_args` within waf.py, which is correctly processing the `Marker`: 

`DEBUG: +++ Marker: AWSLogs/xxxxxxxxx/WAFLogs/us-east-1/2019/10/22
`
Remaining tests conducted on its correct functionality:

### WAF v2 buckets without the -w parameter: 

```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2                           
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ WARNING: No waf_acls parameter found, no log will be fetched
```


### Test with a list of ACLs:


```console
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST-MULTIPLE,AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST-MULTIPLE
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/10/xxxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/15/xxxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Found new log: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ DB Maintenance
root@wazuh-master:/# var/ossec/wodles/aws/aws-s3 --bucket aws-waf-logs-wodle-v2 --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2 -w AWS-WAF-TEST-MULTIPLE,AWS-WAF-TEST1
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST-MULTIPLE
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/10/xxxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST-MULTIPLE/2024/05/03/11/15/xxxxxxxxxx-east-1_AWS-WAF-TEST-MULTIPLE_20240503T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
DEBUG: +++ Working on xxxxxxxxxx - us-east-1 - ACL: AWS-WAF-TEST1
DEBUG: +++ Marker: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/2019/10/22
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/11/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1110Z_a195076a.log.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/xxxxxxxxxx/WAFLogs/us-east-1/AWS-WAF-TEST1/2024/06/04/12/12/xxxxxxxxxx-east-1_AWS-WAF-TEST1_20240604T1115Z_7409afbb.log.gz
DEBUG: +++ No logs to process in bucket: xxxxxxxxxx/us-east-1
DEBUG: +++ DB Maintenance
```


### Kinesis


```console
root@wazuh-master:/# /var/ossec/wodles/aws/aws-s3 --bucket wazuh-aws-wodle-waf  --aws_profile default --only_logs_after 2019-OCT-22 --regions us-east-1 --type waf --skip_on_error -d 2
DEBUG: +++ Debug mode on - Level: 2
DEBUG: No retries configuration found in profile config. Generating default configuration for retries: mode: standard - max_attempts: 10
DEBUG: Created Config object using profile: 'default' configuration
DEBUG: +++ Table does not exist; create
The functionality to process WAF logs stored in S3 via Kinesis was deprecated in 5.0. Consider configuring WAF to store its logs directly in an S3 bucket instead. Check https://documentation.wazuh.com/current/amazon/services/supported-services/waf.html for more information.
DEBUG: +++ Marker: 2019/10/22
DEBUG: ++ Found new log: 2019/10/23/10/aws-waf-logs-delivery-stream-1-2019-10-23-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2019/10/23/11/aws-waf-logs-delivery-stream-1-2019-10-23-11-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake
ERROR: the 2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake file doesn't have the expected structure.
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-1
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-2
DEBUG: ++ Found new log: 2022/11/30/11/aws-waf-logs-delivery-stream-1-2022-11-30-11-32-48-sample-3
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718194
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/05/17/aws-waf-logs-delivery-stream-1-2023-12-05-17-32-48-multiple-values-in-ruleGroupList
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-No-Modifications
DEBUG: ++ Found new log: 2023/12/08/17/aws-waf-logs-delivery-stream-1-2023-12-08-17-32-48-multiple-values-in-ruleGroupList
DEBUG: ++ Found new log: xxxxxxx/2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake
ERROR: the xxxxxxxxx/2022/06/03/aws-waf-logs-delivery-stream-1-2022-06-03-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce4471fake file doesn't have the expected structure.
DEBUG: ++ Found new log: test_prefix/2019/10/22/10/aws-waf-logs-delivery-stream-1-2019-10-22-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718193
ERROR: the test_prefix/2019/10/22/10/aws-waf-logs-delivery-stream-1-2019-10-22-10-32-48-792c6d1f-bfed-4b00-9f5e-88ce44718193 file doesn't have the expected structure.
DEBUG: +++ DB Maintenance
```
